### PR TITLE
feat: add POST request to create a note

### PR DESCRIPTION
**Pull request by LarysaDoroshenko**

### CODE REVIEW

File: src/main/java/com/example/notes/controller/NoteController.java [hunk 1]
```diff
@@ -22,4 +20,10 @@ NoteDtoList getAllNotes() {
         return noteService.getAllNotes();
     }
 
+    @PostMapping("/create")
+    @ResponseStatus(HttpStatus.CREATED)
+    NoteDto createNote(@RequestBody NoteDto noteDto) {
```
Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/controller/NoteController.java]: Replace the NoteDto with the CreateNoteInput dto.
This dto should only contain the content field. The createdAt can be saved inside of the entity on the service layer before the entity is saved.

Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/controller/NoteController.java]: **createNote** can be renamed to **create**.

File: src/main/java/com/example/notes/entity/NoteEntity.java [hunk 2]
```diff
@@ -10,6 +13,9 @@
 @Entity
 @Table(name = "notes")
 @Data
+@Builder
```
Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/entity/NoteEntity.java]: Let us avoid using @Builder with entities for now.

File: src/main/java/com/example/notes/entity/NoteEntity.java [hunk 3]
```diff
@@ -10,6 +13,9 @@
...
+@NoArgsConstructor
+@AllArgsConstructor
```
Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/entity/NoteEntity.java]: These 2 are probably redundant

File: src/main/java/com/example/notes/service/NoteService.java [hunk 4]
```diff
@@ -23,4 +23,10 @@ public NoteDtoList getAllNotes() {
         return new NoteDtoList(noteDtos);
     }
 
+    public NoteDto createNote(NoteDto noteDto) {
```
Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/service/NoteService.java]: **createNote** can be renamed to **create**.

File: src/main/java/com/example/notes/mapper/NoteMapper.java [hunk 5]
```diff
@@ -23,4 +24,12 @@ public NoteDto mapEntityToDto(NoteEntity noteEntity) {
...
     }
 
+    public NoteEntity mapDtoToEntity(NoteDto noteDto) {
+        return NoteEntity.builder()
+                .id(UUID.randomUUID().toString())
+                .content(noteDto.getContent())
+                .createdAt(noteDto.getCreatedAt())
+                .build();
+    }
+
```
Comment (ArzamastsevOleksandr) [inline src/main/java/com/example/notes/mapper/NoteMapper.java]: Perhaps, we do not need this mapper at all at this stage of the application. The mapping logic can be done on the service layer for now. Once the app is large enough and has minor duplications, we can start to extract those to separate components.

**Review by ArzamastsevOleksandr — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/notes/controller/NoteController.java (hunk 1 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/controller/NoteController.java]: Done

Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/controller/NoteController.java]: done

File: src/main/java/com/example/notes/entity/NoteEntity.java (hunk 2 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/entity/NoteEntity.java]: ok

File: src/main/java/com/example/notes/entity/NoteEntity.java (hunk 3 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/entity/NoteEntity.java]: Removed

File: src/main/java/com/example/notes/service/NoteService.java (hunk 4 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/service/NoteService.java]: Done

File: src/main/java/com/example/notes/mapper/NoteMapper.java (hunk 5 above)
Comment (LarysaDoroshenko) [reply src/main/java/com/example/notes/mapper/NoteMapper.java]: Ok

**Review by ArzamastsevOleksandr — APPROVED** [review]

**Pull request merged**